### ISSUE TITLE
`PLE0605` false positive when specifying the generic in `list` or `tuple` constructor

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#6226]
```py
__all__ = list[str]()
```
```
PLE0605 Invalid format for `__all__`, must be `tuple` or `list`
```
